### ISSUE TITLE
Undefined method `option_reader' for Mobility::Backends::ActiveRecord::KeyValue:Class

### ISSUE THREAD

**Issue by bruno-berchielli** [shioyama/mobility#559]
## Context
Ruby 3.1.0
Mobility 1.2.5
Rails 6.1.4 -> Rails 7.0.2.2

When using  `reload!`  inside rails console in a development enviroment with 7.0.2.2 results in:

```
Loading development environment (Rails 7.0.2.2)
[1] pry(main)> reload!
Reloading...
NoMethodError: undefined method `option_reader' for Mobility::Backends::ActiveRecord::KeyValue:Class

        backend_class.option_reader :association_name
                     ^^^^^^^^^^^^^^
from /Users/****/.rvm/gems/ruby-3.1.0/gems/mobility-1.2.5/lib/mobility/backends/key_value.rb:80:in `included'
```

Previously we were using Rails 6.1.4 and upon reverting the rails upgrade, the problem does not happens.

Does not seems to have any other side effect besides the NoMethodError message. 

## Expected Behavior
should reload the codebase without returning errors

**Comment by bruno-berchielli** [comment]
Seems to be related a monkey patch we did to suport JSONB in the KeyValue backend. Closing it as it's not related to the gem itself. Sorry

**Comment by bruno-berchielli** [comment]
Seems to be related to the fact that Rails 7 now defaults the autoloader to Zeitwerk.

Was able to reproduce the same problem in Rails 6 by forcing `config.autoloader = :zeitwerk` instead of `:classic`.

**Comment by shioyama** [comment]
I tried to reproduce this on a fresh Rails 7 application with a single `Post` model with the KeyValue backend and I don't see anything.

```ruby
class Post < ApplicationRecord
  extend Mobility
  translates :title, type: :string
end
```

```
Post.create(title: "foo")
#=> #<Post:0x000055bd71697678 id: 1, author: nil, created_at: Tue, 15 Feb 2022 13:56:56.164298000 UTC +00:00, updated_at: Tue, 15 Feb 2022 13:56:56.167633000 UTC +00:00>
reload!
# Reloading...
#=> true                
```

**Comment by bruno-berchielli** [comment]
It was indeed zeitwerk "fault" but not related to the gem itself. 

We have a monkey patch do extend Mobility KeyValue with Jsonb Fields. When we removed the monkeypatch other errors arised but they were caused by the expectations of `type: :jsonb`. 

This was very hard to debug and was mostly caused by the way Zeitwerk expect to classes/files to be named. Zeitwerk check rake did not catch those differences. 

Closing this as it's not related to the gem itself. 

@shioyama Let me know if you have interest in a PR to support JSONB fields natively with the KeyValue .

Thanks
